packages/integration-karma/test/act/index.spec.js

```diff
@@ -0,0 +1,381 @@
+import UiOutputPercent from 'ui/outputpercent';
+import ForceFoo from 'force/foo';
+import NestedHtmlTags from 'nested/htmlTags';
+import { extractDataIds } from 'test-utils';
+import testProps from './act-components/test-props';
+import testAttrs from './act-components/test-attrs';
+import testBodySlot from './act-components/test-body-slot';
+import testClassAttr from './act-components/test-class-attr';
+import testConditionalFalseAttribute from './act-components/test-conditional-false-attribute';
+import testConditionalTrueAttribute from './act-components/test-conditional-true-attribute';
+import testEmptySlot from './act-components/test-empty-slot';
+import testEmptySlotElementCreation from './act-components/test-empty-slot-element-creation';
+import testHtmlTags from './act-components/test-html-tags';
+import testMultipleChildrenInSlot from './act-components/test-multiple-children-in-slot';
+import testMultipleSlots from './act-components/test-multiple-slots';
+import testNestedHtmlTags from './act-components/test-nested-html-tags';
+import testPropertyReference from './act-components/test-property-reference';
+import testSlotAdjacentToNamedSlot from './act-components/test-slot-adjacent-to-named-slot';
+import testSlotElementCreationWithDuplicateSlotNames from './act-components/test-slot-element-creation-with-duplicate-slot-names';
+import testSlotInGrandchild from './act-components/test-slot-in-grandchild';
+import testStyleAttr from './act-components/test-style-attr';
+
+// Tests that confirm that the runtime LWC engine-dom is compatible with the compiled templates
+// from the ACTCompiler
+describe('ACTCompiler', () => {
+    // These can't be imported from 'lwc'` because @lwc/rollup-plugin won't allow it
+    const { registerComponent, registerTemplate, registerDecorators } = LWC;
+
+    function createComponentFromTemplate(
+        template,
+        { props = {}, propsToTrack = [], methods = {} } = {}
+    ) {
+        const publicProps = {};
+        for (const prop of Object.keys(props)) {
+            if (!propsToTrack.includes(prop)) {
+                publicProps[prop] = { config: 0 };
+            }
+        }
+
+        class CustomElement extends LightningElement {
+            constructor(...args) {
+                super(...args);
+                for (const key of Object.keys(props)) {
+                    this[key] = props[key];
+                }
+            }
+        }
+
+        for (const methodName of Object.keys(methods)) {
+            Object.defineProperty(CustomElement.prototype, methodName, {
+                value: methods[methodName],
+            });
+        }
+
+        const track = {};
+        for (const prop of propsToTrack) {
+            track[prop] = 1;
+        }
+
+        registerDecorators(CustomElement, {
+            publicProps,
+            track,
+        });
+
+        return registerComponent(CustomElement, {
+            tmpl: template,
+        });
+    }
```

Comment:
IIUC the ACTCompiler also has a compiler that creates components as well as templates. I didn't bother to test that, though, since (AIUI) the components are basically dummy components. So I just generate my own dummy components here.